components/definition/cellRules/NetPyNECellRules.js

```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
+      var mechanismsContainer = Object.keys(mechanismsModel).map( mechName => 
+        <NetPyNEMechanismThumbnail 
+          name={mechName} 
+          key={mechName} 
+          selected={mechName == selectedMechanism} 
+          model={mechanismsModel[mechName]} 
+          deleteMethod={this.deleteMechanism}
+          handleClick={this.selectMechanism} 
+        />
+      )
+    }
+    
+    const content = (
+      <CardText className={"tabContainer"} expandable={true}>
+        <div className={"thumbnails"}>
+          <div className="breadcrumb">
+            <IconButton
+              style={styles.home.container}
+              data-tooltip={ page != "main" ? "Go back to main" : undefined }
+              iconStyle={{color: page != 'main' ? cyan500 : pink400, ...styles.home.icon}}
+              onClick={ () => this.setState({page: 'main', selectedCellRule: undefined, selectedSection: undefined, selectedMechanism: undefined})}
+            >
+              <HomeIcon/>
+            </IconButton>
+
+            <NavigationChevronRight style={styles.rightArrow}/>
+
+            <FloatingActionButton
+              zDepth={1}
+              id="newCellRuleButton"
+              style={styles.cellRule}
+              className={"actionButton smallActionButton"}
+              secondary={ page != 'main' }
+              data-tooltip={ 
+                page != 'main'
+                  ? selectedCellRule && selectedCellRule.length > 8 
+                    ? selectedCellRule 
+                    : "Go back to rule"
+                  : "Create rule"
                 }
-                anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
-                targetOrigin={{ horizontal: 'left', vertical: 'top' }}
-              >
-              </IconMenu>
-            </div>
-            <div style={{ clear: "both" }}></div>
-            {sections}
-          </div>
-          <div className={"details"}>
-            {selectedSection}
-            {dialogPop}
-          </div>
-        </CardText>
-      );
+              onClick={() => this.handleHierarchyClick('main')}
+            >
+              { page != 'main' 
+                ? selectedCellRule.length > 8 
+                  ? (selectedCellRule.slice(0,7)+'...') 
+                  : selectedCellRule 
+                : <ContentAdd/>
+              }
+            </FloatingActionButton>
+            
+            <NavigationChevronRight style={styles.rightArrow}/>
+
+            <RaisedButton
+              id="newSectionButton"
+              style={styles.sections.container}
+              disabledBackgroundColor="grey"
+              buttonStyle={styles.sections.icon}
+              primary={ page != 'mechanisms' }
+              secondary={ page == 'mechanisms' }
+              disabled={ selectedCellRule == undefined }
+              onClick={ () => this.handleHierarchyClick('sections') }
+              data-tooltip={ page == 'mechanisms' 
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places? Too difficult to understand what is going on